services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -444,6 +446,54 @@ async def get_project(
                 project_type,
             )
 
+    async def check_project_node_has_all_required_inputs(
+        self, user_id: UserID, project_uuid: ProjectID, node_id: NodeID
+    ) -> None:
+        project_dict, _ = await self.get_project(user_id, f"{project_uuid}")
+        workbench = project_dict["workbench"]
+
+        nodes_map: dict[str, Node] = {i: Node(**workbench[i]) for i in workbench}
```

Comment:
why not item?
```python
nodes_map: dict[str, Node] = {k: Node(**v) for k,v in workbench.item()}
```

Reply:
also why not creating a ```dict[NodeID, Node]``` instead? this would simplify your code and reduce all that stuff with casting all over the place. Also Ruff, pylance help there. Let's get rid of these strings...